src/test/scala/nl.knaw.dans.easy.bagstore/component/BagProcessingSpec.scala

```diff
@@ -200,4 +201,53 @@ class BagProcessingSpec extends BagProcessingFixture {
       }
     }
   }
+
+  "getReferenceBags" should "fail when an empty refbag.txt is found in the bag" in {
+    // first create dir with empty refbag.txt
+    val bagDir = makeBagDirForRefbagTests("")
+
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) should matchPattern {
+      case Failure(e: InvalidBagException) if e == InvalidBagException(bagId, "the bag contains an empty refbags.txt") =>
+    }
+  }
+
+  it should "succeed if no refbag.txt is found in the bag" in {
+    val bagDir = File(testDir.toString) / "refbag"
+    bagDir.createDirectories()
+    bagDir.clear()
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) shouldBe a[Success[_]]
+  }
+
+  it should "fail when an refbag.txt with only whitespaces is found in the bag" in {
+    val bagDir = makeBagDirForRefbagTests("                           ")
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) should matchPattern {
+      case Failure(e: InvalidBagException) if e == InvalidBagException(bagId, "the bag contains an empty refbags.txt") =>
+    }
+  }
+
+  it should "succeed a non-empty refbag.txt is found in the bag" in {
+    val bagDir = makeBagDirForRefbagTests("refbag content")
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) shouldBe a[Success[_]]
+  }
+
+  private def makeBagDirForRefbagTests(content: String): File = {
+    val bagDir = File(testDir.toString) / "refbag"
+    bagDir.createDirectories()
+    bagDir.clear()
```

Comment:
What is there to clear if you just created something? Or can both be the case, a new bag to create or an old to clear? If it is only a new bag than init seems a better name than clear.

Reply:
The clear is done in case there are still lingering files in that directory from previous tests or failed tests

Reply:
The convention in other projects is to clear (drop and create) the testDir in the `before` method. Makes sure this actions doesn't fail when test are playing around with file rights.

Reply:
ok i will move it to the before method